shared-src/objectPrinter/ObjectPrinter.re

```diff
@@ -205,7 +205,7 @@ let base = {
         self.lazy_(self, Obj.magic(o));
       } else if (detectList(~maxLength=maxLength.contents, o)) {
         self.list(self, ~depth, Obj.magic(o));
-      } else if (tag === Obj.first_non_constant_constructor_tag) {
```

Comment:
This seems pretty straightforward, I'm just a little confused by the comment on line 211/what the original intent/idea was and I know that the Obj API is not super well documented. Do you have an understanding of what might have been going on here?

@jordwalke can also probably answer, but I imagine that he is going to be busy for at least the next few days

Reply:
Yes, I found it weird too. But just taking a look to ocaml I can't find any reasoing arround `first_non_constant_constructor_tag`. https://github.com/ocaml/ocaml/blob/5a29ea7c2b31051de814ed7e45bb45098b512b9b/stdlib/obj.ml#L48

```
let [@inline always] is_block a = not (is_int a)
```

Maybe @kyldvs knows more? and possibly the unknown case don't happen anymore?